### PR TITLE
NOJIRA: Refine Gradle tasks that clean Tomcat.

### PR DESCRIPTION
**Pull request by bjagg**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [x] documentation is changed or added

##### Description of change

Improve accuracy of some Gradle tasks.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: gradle/tasks/clean.gradle [hunk 1]
```diff
@@ -1,29 +1,39 @@
...
-   printf "Configured shared libraries from servlet container at location:  ${libs}"
-   FileTree tree = fileTree(dir: libs)
-   delete fileTree(dir:libs , include: '*')
+task cleanShared(type: Delete, dependsOn: 'loadBuildFile') {
+    group 'Clean Tomcat'
+    description 'Removes ALL shared libraries from the servlet container.'
+    followSymlinks = true
+
+    doFirst {
+        def serverbase = props.getProperty("server.home")
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :art: prefer single quotes for constant strings

File: gradle/tasks/clean.gradle [hunk 2]
```diff
@@ -1,29 +1,39 @@
...
-   printf "Configured deployed uPortal from servlet container at location:  ${uPortalDeployDir}"
-   FileTree tree = fileTree(dir: uPortalDeployDir)
-   delete uPortalDeployDir
+task cleanPortal(type: Delete, dependsOn: 'loadBuildFile') {
+    group 'Clean Tomcat'
+    description 'Removes the deployed uPortal from the servlet container.'
+    followSymlinks = true
+
+    doFirst {
+        def serverbase = props.getProperty("server.home")
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :art: prefer single quotes for constant strings

File: gradle/tasks/clean.gradle [hunk 3]
```diff
@@ -1,29 +1,39 @@
...
 
 task loadBuildFile() {
-    println "Load the build.properties file"
-    def buildPropsFile = file('build.properties')
-    if (buildPropsFile.exists()) {
-        buildPropsFile.withInputStream { props.load(it) }
-    } else {
-        println("WARNING: build.properties not found")
+    doLast {
+        println "Load the build.properties file"
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :art: prefer single quotes for constant strings

File: gradle/tasks/clean.gradle [hunk 4]
```diff
@@ -1,29 +1,39 @@
...
+        def buildPropsFile = file('build.properties')
+        if (buildPropsFile.exists()) {
+            buildPropsFile.withInputStream {props.load(it)}
+        } else {
+            println("WARNING: build.properties not found")
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :art: prefer single quotes for constant strings

File: gradle/tasks/hsql.gradle [hunk 5]
```diff
@@ -15,12 +15,14 @@ import com.wiredforcode.gradle.spawn.*
 
 // NOTE: The startHsql task can be executed after uportal-war is built because it uses hsqldb-2.3.2.jar in classpath, otherwise it will fail
 task startHsql(type: SpawnProcessTask) {
-    description "Start hsql server"
-    def cmd = "java -cp ./uportal-war/target/uPortal/WEB-INF/lib/hsqldb-2.3.2.jar org.hsqldb.server.Server --database.0 file:./data/uPortal;hsqldb.tx=mvcc --dbname.0 uPortal --address localhost --port 8887"
+    group 'Hypersonic SQL'
+    description "Start hsql server."
```
Comment (ChristianMurphy) [inline gradle/tasks/hsql.gradle]: :art: prefer single quotes for constant strings

File: gradle/tasks/hsql.gradle [hunk 6]
```diff
@@ -15,12 +15,14 @@ import com.wiredforcode.gradle.spawn.*
...
+    def cmd = "java -cp ${projectDir}/uportal-war/target/uPortal/WEB-INF/lib/hsqldb-2.3.2.jar org.hsqldb.server.Server --database.0 file:./data/uPortal;hsqldb.tx=mvcc --dbname.0 uPortal --address localhost --port 8887"
     command cmd
     ready 'Startup sequence completed'
 }
 
-task stopHsql(type: KillProcessTask){
-    description "Stop hsql server"
+task stopHsql(type: KillProcessTask) {
+    group 'Hypersonic SQL'
+    description "Stop hsql server."
```
Comment (ChristianMurphy) [inline gradle/tasks/hsql.gradle]: :art: prefer single quotes for constant strings

File: gradle/tasks/clean.gradle [hunk 7]
```diff
@@ -1,29 +1,39 @@
 def props = new Properties()
 
 task loadBuildFile() {
-    println "Load the build.properties file"
-    def buildPropsFile = file('build.properties')
-    if (buildPropsFile.exists()) {
-        buildPropsFile.withInputStream { props.load(it) }
-    } else {
-        println("WARNING: build.properties not found")
+    doLast {
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :question: why is `doLast` needed here?

File: gradle/tasks/clean.gradle [hunk 8]
```diff
@@ -1,29 +1,39 @@
...
-   def libs =  "${serverbase}/shared/lib"
-   printf "Configured shared libraries from servlet container at location:  ${libs}"
-   FileTree tree = fileTree(dir: libs)
-   delete fileTree(dir:libs , include: '*')
+task cleanShared(type: Delete, dependsOn: 'loadBuildFile') {
+    group 'Clean Tomcat'
+    description 'Removes ALL shared libraries from the servlet container.'
+    followSymlinks = true
+
+    doFirst {
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :question: why is `doFirst` needed here?

File: gradle/tasks/clean.gradle [hunk 9]
```diff
@@ -1,29 +1,39 @@
...
-   def uPortalDeployDir = "${serverbase}/webapps/uPortal"
-   printf "Configured deployed uPortal from servlet container at location:  ${uPortalDeployDir}"
-   FileTree tree = fileTree(dir: uPortalDeployDir)
-   delete uPortalDeployDir
+task cleanPortal(type: Delete, dependsOn: 'loadBuildFile') {
+    group 'Clean Tomcat'
+    description 'Removes the deployed uPortal from the servlet container.'
+    followSymlinks = true
+
+    doFirst {
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :question: why is `doFirst` needed here?

File: gradle/tasks/clean.gradle [hunk 10]
```diff
@@ -1,29 +1,39 @@
...
 }
 
-task cleanTomcat(type:Delete) {
-   description 'Removes the deployed uPortal from the servlet container.'
-   def serverbase = props.getProperty("server.home")
-   def uPortalDeployDir = "${serverbase}/webapps/uPortal"
-   printf "Configured deployed uPortal from servlet container at location:  ${uPortalDeployDir}"
-   FileTree tree = fileTree(dir: uPortalDeployDir)
-   delete uPortalDeployDir
+task cleanPortal(type: Delete, dependsOn: 'loadBuildFile') {
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :+1:

File: gradle/tasks/clean.gradle [hunk 11]
```diff
@@ -1,29 +1,39 @@
...
 }
 
-task cleanShared(type:Delete) {
-   description 'Removes ALL shared libraries from the servlet container.'
-   def serverbase = props.getProperty("server.home")
-   def libs =  "${serverbase}/shared/lib"
-   printf "Configured shared libraries from servlet container at location:  ${libs}"
-   FileTree tree = fileTree(dir: libs)
-   delete fileTree(dir:libs , include: '*')
+task cleanShared(type: Delete, dependsOn: 'loadBuildFile') {
```
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :+1:

File: gradle/tasks/clean.gradle (hunk 4 above)
Comment (ChristianMurphy) [inline gradle/tasks/clean.gradle]: :art: parens are unneeded

File: gradle/tasks/clean.gradle (hunk 7 above)
Comment (bjagg) [inline gradle/tasks/clean.gradle]: Typically, tasks define clojures. See https://docs.gradle.org/3.5/userguide/tutorial_using_tasks.html

File: gradle/tasks/clean.gradle (hunk 8 above)
Comment (bjagg) [inline gradle/tasks/clean.gradle]: The delete string needs to be set before the core work, deleting the files/directories, is performed. doLast{} does not set this variable in time.

File: gradle/tasks/clean.gradle (hunk 9 above)
Comment (bjagg) [inline gradle/tasks/clean.gradle]: The delete string needs to be set before the core work, deleting the files/directories, is performed. doLast{} does not set this variable in time.

File: gradle/tasks/hsql.gradle [hunk 12]
```diff
@@ -15,12 +15,14 @@ import com.wiredforcode.gradle.spawn.*
 
 // NOTE: The startHsql task can be executed after uportal-war is built because it uses hsqldb-2.3.2.jar in classpath, otherwise it will fail
 task startHsql(type: SpawnProcessTask) {
-    description "Start hsql server"
-    def cmd = "java -cp ./uportal-war/target/uPortal/WEB-INF/lib/hsqldb-2.3.2.jar org.hsqldb.server.Server --database.0 file:./data/uPortal;hsqldb.tx=mvcc --dbname.0 uPortal --address localhost --port 8887"
+    group 'Hypersonic SQL'
+    description 'Start hsql server.'
+    def cmd = "java -cp ${projectDir}/uportal-war/target/uPortal/WEB-INF/lib/hsqldb-2.3.2.jar org.hsqldb.server.Server --database.0 file:./data/uPortal;hsqldb.tx=mvcc --dbname.0 uPortal --address localhost --port 8887"
```
Comment (ChristianMurphy) [inline gradle/tasks/hsql.gradle]: :question: should [`projectDir`](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:projectDir) be [`rootDir`](https://docs.gradle.org/current/dsl/org.gradle.api.Project.html#org.gradle.api.Project:rootDir)?

Comment (bjagg) [inline gradle/tasks/hsql.gradle]: In this case, the task is meant to be run from the root project. With the coming changes that may move this around, I will change this to rootDir -- thanks for finding this one!

**Pull request merged**